### ISSUE TITLE
Erroneous Shadow Caster Culling when FilterManager is Enabled

### ISSUE THREAD

**Issue by Moguri** [panda3d/panda3d#1036]
## Description
As soon a FilterManager is used, shadows stop working for me using a `master` build (1.10.x works fine). This has been tracked down to the shadow caster getting culled due to d11d4e0d85a18597811d18ea33a250d2f7bf6c7f.

## Steps to Reproduce
Run the following Python program
```python
import panda3d.core as p3d
from direct.showbase.ShowBase import ShowBase
from direct.filter.CommonFilters import CommonFilters

class App(ShowBase):
    def __init__(self):
        super().__init__()

        self.render.set_shader_auto()

        # Enable FilterManager
        self.filters = CommonFilters(self.win, self.cam)
        # Comment out the below line to bring shadows back
        self.filters.setHighDynamicRange()

        # Add something to cast onto
        self.env = self.loader.load_model('environment.egg')
        self.env.reparent_to(self.render)

        # Add something to cast a shadow
        self.model = self.loader.load_model('smiley.egg')
        self.model.reparent_to(self.render)
        self.model.set_pos(0, 0, 3)

        # Position camera to see model and shadow
        self.disableMouse()
        self.camera.set_pos((10, 10, 10))
        self.camera.look_at(self.model)

        # Add a shadow casting spotlight
        self.light = p3d.Spotlight('caster')
        self.light.set_shadow_caster(True, 513, 512)
        self.lightnp = self.render.attach_new_node(self.light)
        self.lightnp.set_pos((5, -5, 10))
        self.lightnp.look_at(self.model)
        self.render.set_light(self.lightnp)

APP = App()
APP.run()
```

## Environment
 * Operating system: Arch Linux (5.8.10 kernel)
 * System architecture: x86_64
 * Panda3D version: 1.11 build from `master` branch (eb367430f7d4aad7d01e5b9212534b066e5a21f6)
 * Installation method: built from source
 * Python version (if using Python): 3.8.5
 * Compiler (if using C++): Clang 10.0.1

**Comment by rdb** [comment]
This reminds me an awful lot of #890 and suspect it may be related.

**Comment by rdb** [comment]
Actually, it looks like this is a sorting issue.  FilterManager is forcing the render of the scene to come before the render of the shadow cameras, so at the time that the main scene is culled, the cull system assumes that no shadow caster is active.

This also explains why in 1.10 it shows the scene unshadowed for a frame.

I'm a little on the fence about whether this should be considered an issue in the cull system, or just in FilterManager for forcing it to sort before everything else.

**Comment by rdb** [comment]
The default FilterManager sort is normally set to -1000.
The default shadow buffer sort is -10.

A workaround would be to change any shadow casters to have a sort value smaller than -1000.  This is what I would recommend to also fix the 1.10 issue wherein the scene is unshadowed for a single frame.

Since the shadow buffer sort is baked into bam files, I have no choice but to change the default FilterManager sort to -9.  I'm changing this only on the master branch for now.  It does mean that there is effectively a limit of 8 postprocessing stages.

The long term solution is to get rid of the buffer sorting system and instead replace it with a dependency graph-based approach.
